Formula/telnetd.rb

```diff
@@ -0,0 +1,48 @@
+class Telnetd < Formula
+  desc "TELNET server (built from macOS Sierra sources)"
+  homepage "https://opensource.apple.com/"
+  url "https://opensource.apple.com/tarballs/remote_cmds/remote_cmds-54.50.1.tar.gz"
+  sha256 "156ddec946c81af1cbbad5cc6e601135245f7300d134a239cda45ff5efd75930"
+
+  keg_only :provided_pre_high_sierra
+
+  depends_on :xcode => :build
+
+  resource "libtelnet" do
+    url "https://opensource.apple.com/tarballs/libtelnet/libtelnet-13.tar.gz"
+    sha256 "e7d203083c2d9fa363da4cc4b7377d4a18f8a6f569b9bcf58f97255941a2ebd1"
+  end
+
+  def install
+    resource("libtelnet").stage do
+      xcodebuild "SYMROOT=build"
+
+      libtelnet_dst = buildpath/"telnetd.tproj/build/Products"
+      libtelnet_dst.install "build/Release/libtelnet.a"
+      libtelnet_dst.install "build/Release/usr/local/include/libtelnet/"
+    end
+
+    system "make",
+           "-C", "telnetd.tproj",
+           "OBJROOT=build/Intermediates",
+           "SYMROOT=build/Products",
+           "DSTROOT=build/Archive",
+           "CFLAGS=$(CC_Flags) -isystembuild/Products/",
+           "LDFLAGS=$(LD_Flags) -Lbuild/Products/"
+
+    bin.install "telnetd.tproj/build/Products/telnetd"
+    man8.install "telnetd.tproj/telnetd.8"
+  end
+
+  test do
+    # running a whole server, connecting, and so forth is a bit clunky and hard
```

Comment:
Per an earlier comment:

```
What about if you use 2>&1 at the end of that shell_output line.

assert_match "usage: telnetd", shell_output("#{bin}/telnetd usage 2>&1")
```

Reply:
@DomT4 That does not work:

```sh
$ brew test telnetd
Testing telnetd
==> /usr/local/Cellar/telnetd/54.50.1/bin/telnetd usage 2>&1
Error: telnetd: failed
<0> expected but was
<1>.
```

Reply:
Off the top of my head, I don't think `PTY` checks exit status, so it may well already be chucking out a bad one & it's simply not getting noticed. `shell_output()` always checks exit status. You can tell it to expect a bad one, and as long as the `assert_match` is still true it won't blow up.

```ruby
assert_match "usage: telnetd", shell_output("#{bin}/telnetd usage 2>&1", 1)
```

Reply:
@DomT4 It is irrelevant if `PTY` checks exit status. Running `telnetd usage` already returns a non-zero exit status by default. Example:

```sh
$ telnetd usage
usage: telnetd [-debug] [-D (options|report|exercise|netdata|ptydata)]
	 [-h] [-k] [-l] [-n]
	 [-u utmp_hostname_length] [-U] [port]
$ echo $?
1
$ 
```

The only thing that matters in this test is that `telnetd usage` returns the expected output. This is sufficient to determine whether the compilation succeeded.

Reply:
> This is sufficient to determine whether the compilation succeeded.

Sure, but equally as a broad rule the simplest possible code is ideal in `homebrew/core` because `homebrew/core` is intended to be _pretty_ friendly to people new to Ruby. The more code drifts from the most common usage the less likely people are to feel able to touch it in future, again, speaking very broadly.

Reply:
@DomT4 Sure, I understand that. I'm not sure how to change this code in order to be friendly like that while still making it functional, though...

Reply:
```diff
diff --git a/Formula/telnetd.rb b/Formula/telnetd.rb
index b27c1e4f41..ef313c4913 100644
--- a/Formula/telnetd.rb
+++ b/Formula/telnetd.rb
@@ -35,14 +35,6 @@ class Telnetd < Formula
   end
 
   test do
-    # running a whole server, connecting, and so forth is a bit clunky and hard
-    # to write properly so...
-    require "pty"
-    require "expect"
-
-    PTY.spawn "#{bin}/telnetd usage" do |input, _output, _pid|
-      str = input.expect(/usage: telnetd/)
-      assert_match "usage: telnetd", str[0]
-    end
+    assert_match "usage: telnetd", shell_output("#{bin}/telnetd usage 2>&1", 1)
   end
 end
```

Does the job. Tested locally.

Reply:
@DomT4 Ok, that works for `telnetd`. I thought I recalled it not working for some reason, but I guess not. I'll apply your patch, although I tested the same sort of thing for `tnftp` and `tnftpd` and they did not work without `PTY`.

Reply:
Yeah, that's expected. You're poking `tnftp` with faux-interactivity, whereas this one is just doing a straight string read. Not sure what's going on with `tnftpd` - It looks like it should work with this format of test too.

Reply:
@DomT4 Yes, I expected the same WRT `tnftpd`. 🤷‍♂️